src/View/Helper/CommonModulesHelper.php

```diff
@@ -88,7 +88,14 @@ public function createFilterByLangMod($maxNumberOfParams = 1)
             if (isset($params[$maxNumberOfParams-1])) {
                 $lang  = $params[$maxNumberOfParams-1];
             }
-
+            
+            if ($this->request->params['controller'] == 'Vocabulary' && $this->request->params['action'] == 'add_sentences') {
+                $filteredLang = $this->request->getSession()->read('vocabulary_requests_filtered_lang');
+                if ($filteredLang != null && $lang == 'und') {
+                    $lang = $filteredLang;
+                }
+            }
+            
```

Comment:
I think this code is not needed any more because the preselected value of the dropdown can be now directly set from the URL.